ydb/docs/en/core/concepts/optimizer.md

```diff
@@ -0,0 +1,81 @@
+# Query Optimization in {{ ydb-short-name }}
+
+{{ ydb-short-name }} uses two types of query optimizers: a rule-based optimizer and a cost-based optimizer. The cost-based optimizer is used for complex queries, usually analytical ([OLAP](https://en.wikipedia.org/wiki/Online_analytical_processing)), while rule-based optimization works on all queries.
+
+A query plan is a graph of operations, such as reading data from a source, filtering a data stream by a predicate, or more complex operations such as [JOIN](../yql/reference/syntax/join.md) и [GROUP BY](../yql/reference/syntax/group_by.md). Optimizers in YDB take an initial query plan as input and transform it into a more efficient plan that is equivalent to the initial one in terms of the returned result.
+
+## Rule-based Optimizer
+
+A significant part of the optimizations in YDB is applicable to almost any query plan, which eliminates the need to analyze alternative plans and their costs. The rule-based optimizer consists of a set of heuristic rules that are applied whenever possible. For example, it is useful to filter out data as early as possible in the execution plan for any query. Each optimizer rule comprises a condition that triggers the rule and rewriting logic that is executed when the plan is applied. Rules are applied iteratively as long as any rule conditions match.
+
+## Cost-Based Query Optimizer
+
+Cost-Based Optimizer is used for more complex optimizations, such as choosing an optimal join order and join algorithms. Cost-based Optimizer considers a large number of alternative execution plans for each query and chooses the best one based on the cost estimate for each option. Currently, this optimizer only works with plans that contain [JOIN](../yql/reference/syntax/join.md) operations. It chooses the best order for these operations to be executed, and also chooses the most efficient implementation for each Join algoritm in the plan.
+
+The cost optimizer consists of three main components:
+
+* Plan enumerator
+* Cost estimation function
+* Statistics module, which is used to estimate statistics for the cost function
+
+### Plan enumerator
+
+The current Cost-Based Optimizer in {{ ydb-short-name }} enumerates all useful join trees, for which the join conditions are defined. We first build a join hypergraph, where the nodes are tables and edges are join conditions. Depending on how the original query was written, the join hypergraph may have quite different topologies, varying from simple chain-like graphs to complex cliques. The resulting topology of the join graph determines how many possible alternite plans need to be considered by the optimizer.
+
+For example, a star is a common topology in analytical queries, where a main fact table is joined to multiple dimension tables:
```

Comment:
```suggestion
### Plan enumerator

The current cost-based optimizer in {{ ydb-short-name }} enumerates all useful join trees for which the join conditions are defined. First, it builds a join hypergraph, where the nodes represent tables and the edges represent join conditions. Depending on how the original query is written, the join hypergraph can exhibit various topologies, ranging from simple chain-like graphs to complex cliques. The resulting topology of the join graph determines how many potential alternative plans the optimizer must consider.

For example, a star is a common topology in analytical queries, where a main fact table is joined to multiple dimension tables:
```

Reply:
I don't see any reasoning why you think your version is better (except correcting capitalization and a typo)